R/type2.R

```diff
@@ -409,3 +422,41 @@ vec_is_subtype <- function(x, super, ..., x_arg = "", super_arg = "") {
 vec_implements_ptype2 <- function(x) {
   .Call(vctrs_implements_ptype2, x)
 }
+
+# Example usage of ptype2 restart. As there is no cast restart (yet),
+# it requires `x` and `y` to be convertible to the restarted ptype.
+with_ordered_restart <- function(expr) {
+  withCallingHandlers(
+    expr,
+    vctrs_error_ptype2 = function(cnd) {
+      x <- cnd[["x"]]
+      y <- cnd[["y"]]
+
+      restart <- FALSE
+
+      if (is.ordered(x)) {
```

Comment:
I guess this allows inheritance back into the ptype2 hierarchy. i.e. if someone made a ordered subclass it could hit this. Is `inherits_only(x, c("ordered", "factor"))` better?

Reply:
It depends on the goals of this method. For ggplot I think you'd want inheritance to be maximally compatible.